### PR TITLE
Add healthcheck endpoint

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Add `GET /healthcheck` endpoint
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes [#33218](https://github.com/PrestaShop/PrestaShop/issues/33218)
| Sponsor company   | PrestaShop SA
| How to test?      | Go to `/healthcheck`

Capture d’écran 2023-07-26 à 16 39 56

### CODE REVIEW

**Comment by JMorinPS** [comment]
Once the CICD repository is up to date, we should be able to trigger integration deployment on this PR

**Comment by JMorinPS** [comment]
@boherm is this PR coming from a fork ? If so, we won't be able to trigger any deployment, repository secrets won't be available from a fork branch

**Comment by boherm** [comment]
Hi @JMorinPS, yep this PR coming from a fork so we can't trigger deployment action.
Maybe, we can make a new branch directly in `PrestaShop/kanbanbot` or just merge this PR into main. 🤔

**Comment by JMorinPS** [comment]
This could be our best pattern :+1:

File: src/Shared/Controller/HealthcheckController.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private function checkGithubAccess(): bool
+    {
+        try {
+            $this->pullRequestRepository->find(
+                new PullRequestId('PrestaShop', 'PrestaShop', '1')
+            );
+
+            return true;
+        } catch (\Exception $ex) {
+        }
```
Comment (M0rgan01) [inline src/Shared/Controller/HealthcheckController.php:64]: What error can be raised here?

Comment (boherm) [reply src/Shared/Controller/HealthcheckController.php:64]: Here, it really don't matter. At any exceptions, we must say that the github link is down.

**Review by M0rgan01 — APPROVED** [review]

**Review by matks — COMMENTED** [review]
I am sorry @boherm I reviewed the wrong PR 😅 please look at my comments in https://github.com/PrestaShop/kanbanbot/pull/17

Note: these comments can be done on another PR if you prefer ;) so we could merge this one and submit another

**Comment by boherm** [comment]
@matks, I've push some tests :)
I agree on your other comments that can could be done on another PR.

**Review by matks — APPROVED** [review]

**Pull request merged**